src/main/java/org/spongepowered/api/event/cause/entity/damage/source/DamageSources.java

```diff
@@ -63,6 +65,12 @@
 
     public static final DamageSource VOID = DummyObjectProvider.createFor(DamageSource.class, "VOID");
 
+    /**
+     * Generally used to describe the damage taken when colliding into a wall
+     * while flying with an elytra.
+     */
+    public static final DamageSource WALL_COLLISION = DummyObjectProvider.createFor(DamageSource.class, "WALL_COLLISION");
```

Comment:
I'm not fond of calling wall collision to be a raw DamageSource, maybe the `DamageType` supported for wall collisions, but I don't think we should be using just a vanilla `DamageSource` when we can provide some more information about the `DamageSource` (like, do we want to use `BlockDamageSource` or `EntityDamageSource`?)

Reply:
And use the block that they collide into?